astroid/nodes/node_classes.py

```diff
@@ -910,15 +921,16 @@ def default_value(self, argname):
         :raises NoDefault: If there is no default value defined for the
             given argument.
         """
-        args = self.arguments
+        # Ignore *args and **kwargs
+        args = list(filter(lambda x: not isinstance(x, Name), self.arguments))
+        index = _find_arg(argname, self.kwonlyargs)[0]
+        if index is not None and self.kw_defaults[index] is not None:
+            return self.kw_defaults[index]
```

Comment:
I'm wondering if we can avoid moving this code around and also avoid filtering out *args and **kwargs by just doing a better lookup in self.defaults instead of doing math with len(). I hope this might help you solve the test case I provided.

Reply:
I have implemented a new method, it works based on column offsets, I think it should give the right answer every time as long as the argument and the default value are declared on the same line (I think this is true for practically all code in existence?). 

What do you think?

Reply:
It feels too complex for the problem we're trying to solve. I would have thought something like this would work:

```diff
diff --git a/astroid/nodes/node_classes.py b/astroid/nodes/node_classes.py
index 910c00a4..4d2eae2f 100644
--- a/astroid/nodes/node_classes.py
+++ b/astroid/nodes/node_classes.py
@@ -928,9 +928,12 @@ class Arguments(_base_nodes.AssignTypeNode):
             return self.kw_defaults[index]
         index = _find_arg(argname, args)[0]
         if index is not None:
-            idx = index - (len(args) - len(self.defaults))
+            idx = index - (len(args) - len(self.defaults) - len(self.kw_defaults))
             if idx >= 0:
                 return self.defaults[idx]
+        index = _find_arg(argname, self.kwonlyargs)[0]
+        if index is not None and self.kw_defaults[index] is not None:
+            return self.kw_defaults[index]
         raise NoDefault(func=self.parent, name=argname)
 
     def is_argument(self, name) -> bool:
```

Reply:
Just to make sure, this is how you envisioned the code?

```python
args = self.arguments
index = _find_arg(argname, self.kwonlyargs)[0]
if index is not None and self.kw_defaults[index] is not None:
    return self.kw_defaults[index]
index = _find_arg(argname, args)[0]
if index is not None:
    idx = index - (len(args) - len(self.defaults) - len(self.kw_defaults))
    if idx >= 0:
        return self.defaults[idx]
index = _find_arg(argname, self.kwonlyargs)[0]
if index is not None and self.kw_defaults[index] is not None:
    return self.kw_defaults[index]
raise NoDefault(func=self.parent, name=argname)
```

With that code 4 tests are failing, I can probably work something out, however I am having a hard time what the subtraction means or tries to achieve? That is why I went with the brute force method of looping through everything, maybe once I understand what the rationale behind subtraction is, I will have an easier time modifying it.

Reply:
Ah, I also meant to keep the line that filters out *args and **kwargs. I think tests were passing with that.

(In general, I'd suggest not force pushing during an active review; we'll squash merge at the end or you can rebase yourself once the review is done.)

Reply:
Got it. I updated all points of the review. Does this mean that we will keep the filter *args, **kwargs methodology?

Reply:
Yes, so it seems to me at the moment. Thanks!